flare_portal/experiments/views.py

```diff
@@ -213,6 +214,36 @@ def form_valid(self, form: ParticipantBatchForm) -> HttpResponse:  # type: ignor
 participant_create_batch_view = ParticipantCreateBatchView.as_view()
 
 
+class ParticipantUploadView(FormView):
+    form_class = ParticipantUploadForm
+    template_name = "experiments/participant_upload_form.html"
+
+    def get_success_url(self) -> str:
+        return reverse(
+            "experiments:participant_list",
+            kwargs={
+                "project_pk": self.kwargs["project_pk"],
+                "experiment_pk": self.kwargs["experiment_pk"],
+            },
+        )
+
+    def dispatch(self, *args: Any, **kwargs: Any) -> HttpResponse:
+        self.experiment = get_object_or_404(Experiment, pk=kwargs["experiment_pk"])
+        return super().dispatch(*args, **kwargs)
+
+    def get_context_data(self, **kwargs: Any) -> dict:
+        context = super().get_context_data(**kwargs)
+        context["experiment"] = self.experiment
+        return context
+
+    def form_valid(self, form: ParticipantUploadForm) -> HttpResponse:  # type: ignore
+        form.save(experiment=self.experiment)
```

Comment:
It would be nice to show a success message after the upload succeeds.

Reply:
Added!